robot/impl/local_robot.go

```diff
@@ -709,12 +723,135 @@ func (r *localRobot) updateWeakDependents(ctx context.Context) {
 	}
 }
 
-// FrameSystemConfig returns the info of each individual part that makes up a robot's frame system.
-func (r *localRobot) FrameSystemConfig(
-	ctx context.Context,
-	additionalTransforms []*referenceframe.LinkInFrame,
-) (framesystemparts.Parts, error) {
-	return r.frameSvc.Config(ctx, additionalTransforms)
+// Config returns the info of each individual part that makes up the frame system
+// The output of this function is to be sent over GRPC to the client, so the client
+// can build its frame system. requests the remote components from the remote's frame system service.
+// NOTE(RDK-258): If remotes can trigger a local robot to reconfigure, you don't need to update remotes in every call.
+func (r *localRobot) FrameSystemConfig(ctx context.Context) (*framesystem.Config, error) {
+	r.fsMu.RLock()
```

Comment:
What's the mutex needed for? I don't see the race condition

Reply:
I think this is leftover from code I copied over and can probably be removed safely now that the way things are being called is a lot cleaner.

Reply:
I think the existing `mu` should guard well enough.